stages/org.osbuild.zipl.inst

```diff
@@ -74,8 +82,8 @@ def find_kernel(root, kernel: str):
         if version != kernel:
             continue
 
-        linux = root + params["linux"]
-        initrd = root + params["initrd"]
+        linux = f"{root}/boot" + params["linux"]
+        initrd = f"{root}/boot" + params["initrd"]
```

Comment:
ok I'm thinking we shouldn't make this change here, but rather below let's do something like this here:

```suggestion
        linux = params["linux"]
        initrd = params["initrd"]
```

and then below:

```
kernel, initrd, kopts = find_kernel(root, kver)                     
# Build up the full path to the kernel/initrd. If boot is a separate
# filesystem then add /boot in the path.                            
if os.path.ismount(os.path.join(root, "boot")):                     
    kernel = os.path.join(root, "boot", kernel)                     
    initrd = os.path.join(root, "boot", initrd)                     
else:                                                               
    kernel = os.path.join(root, kernel)                             
    initrd = os.path.join(root, initrd)                             

```

Reply:
Maybe this should be a separate PR too.

Reply:
so something like this:

```
        # Find the prefix that needs to be added to get the full path to 
        # the kernel/initrd. If /boot is a separate filesystem then the  
        # path inside the blscfg won't have that prefix on it and we need
        # to add it here.                                                
        prefix = root                                                    
        if os.path.ismount(os.path.join(root, "boot")):                  
            prefix = os.path.join(root, "boot")                          
                                                                         
        linux = os.path.join(prefix, params["linux"])                    
        initrd = os.path.join(prefix, params["initrd"])                  
        options = params.get("options", "")                              
        return linux, initrd, options                                    

```

Reply:
updated the above example to add `.lstrip("/")` - otherwise `os.path.join()` will just use the last parameter.